lib/Service/DownloadModelsService.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace OCA\Recognize\Service;
+
+use OCA\Recognize\Helper\TAR;
+
+class DownloadModelsService {
+
+    /**
+     * @return void
+     * @throws \Exception
+     */
+    public function download() : void {
+        $targetPath = __DIR__ . '/../../models';
+        if (file_exists($targetPath)) {
+            return;
+        }
+
+        $archiveURL = $this->getArchiveUrl($this->getNeededArchiveRef());
+        $archivePath = __DIR__ . '/../../models.tar.gz';
+        $read = fopen($archiveURL, 'r');
+        $write = fopen($archivePath, 'w');
+        stream_copy_to_stream($read, $write);
+        $tarManager = new TAR($archivePath);
+        $files = $tarManager->getFiles();
+        $mainFolder = $files[0];
+        $modelFolder = $mainFolder . 'models';
+        $modelFiles = array_filter($files, function($path) use ($modelFolder) {
+            return str_starts_with($path, $modelFolder . '/') || str_starts_with($path, $modelFolder);
```

Comment:
At least with server we currently still cover PHP 7.4 and str_starts_with is only available with PHP8.0 or later.

Reply:
Huh, weird, I believe I tested with php7.4 (and PHPStorm also didn't blink :( )